conformance/tests/directives_deprecated_protocol.py

```diff
@@ -0,0 +1,34 @@
+"""
+Tests the warnings.deprecated function when it is used in a Protocol.
+"""
+
+# > There are additional scenarios where deprecations could come into play.
+# > For example, an object may implement a `typing.Protocol`,
+# > but one of the methods required for protocol compliance is deprecated.
+# > As scenarios such as this one appear complex and relatively unlikely to come up in practice,
+# > this PEP does not mandate that type checkers detect them.
+
+from typing import Protocol
+
+from typing_extensions import deprecated
+
+
+class Fooable(Protocol):
+
+    @deprecated("Deprecated")
+    def foo(self) -> None: ...
+
+    def bar(self) -> None: ...
+
+
+class Fooer(Fooable):
+
+    def foo(self) -> None:  # E?: Implementation of deprecated method foo
```

Comment:
I don't think this should be a warning at all, unless it is marked with `@override`.

The scenario envisioned in the quoted text is actually a little different and more convoluted, something like this:

```
class Fooable(Protocol):
    def foo(self) -> None: ...

class Concrete:
    @deprecated("Deprecated")
    def foo(self) -> None: ...

def take_fooable(f: Fooable) -> None: ...

def caller(c: Concrete) -> None:
    take_fooable(c)  # Concrete is a Fooable, but only because of a deprecated method
```